src/Sentry.Unity/TimeDebounceBase.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+
+namespace Sentry.Unity
+{
+    public interface IUnityLogMessageDebounce
+    {
+        bool Debounced();
+    }
+
+    /// <summary>
+    /// This class is not thread-safe and is designed to be called by Unity non-threaded logger callback
+    /// </summary>
+    internal class TimeDebounceBase : IUnityLogMessageDebounce
+    {
+        private static DateTimeOffset Now => DateTimeOffset.UtcNow;
+
+        protected TimeSpan _debounceOffset;
+
+        private DateTimeOffset? _barrierOffset;
+
+        public bool Debounced()
+        {
+            if (_barrierOffset != null && Now < _barrierOffset)
+            {
+                return false;
+            }
+
+            _barrierOffset = Now.Add(_debounceOffset);
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// This class is not thread-safe and is designed to be called by Unity non-threaded logger callback
+    /// </summary>
+    internal sealed class LogTimeDebounce : TimeDebounceBase
+    {
+        public LogTimeDebounce() => _debounceOffset = TimeSpan.FromSeconds(1);

```

Comment:
For now this is fine but these intervals will need to be configurable.
It'll be simpler to have a single type that takes an interval via the ctor.

Reply:
100%! My thoughts too. Need to decide what value do we expect (millis, seconds etc)